### ISSUE TITLE
[sigh] recreates the new AdHoc profile in Apple Developer with different names (like "cn.xxxx AdHoc 1544776407"), when we updated Fastlane to 2.111.0

### ISSUE THREAD

**Issue by UZheng** [fastlane/fastlane#13936]
- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

It is a new issue from [#13910](https://github.com/fastlane/fastlane/issues/13910).

Everything works fine before this error report for "Can't find class 'adhoc'([#13905](https://github.com/fastlane/fastlane/issues/13905)).Then I solved the problem according to the instructions to upgrade the version to 2.111.0, but this situation has occurred. 
I replace some important information with some characters, ignore it please.

##### Complete output when running fastlane, including the stack trace and command used

<details>

  <pre>
+ fastlane for_pgyAdhoc  'des: some infomation' isDebug:true

[19:24:14]: Your Fastfile has had smart quotes sanitised. To avoid issues in the future, you should not use TextEdit for editing it. If you are not using TextEdit, you should turn off smart quotes in your editor of choice.
[19:24:14]: ------------------------------
[19:24:14]: --- Step: default_platform ---
[19:24:14]: ------------------------------
[19:24:14]: Driving the lane 'ios start' 🚀
[19:24:14]: ---------------------------
[19:24:14]: --- Step: git branch -l ---
[19:24:14]: ---------------------------
[19:24:14]: $ git branch -l
[19:24:14]: ▸ * (HEAD detached at d2b11c733)
[19:24:14]: ------------------------
[19:24:14]: --- Step: git_branch ---
[19:24:14]: ------------------------
[19:24:14]: -----------------------------
[19:24:14]: --- Step: last_git_commit ---
[19:24:14]: -----------------------------
[19:24:14]: $ 开始执行--> build 
[19:24:14]: --------------------------------------
[19:24:14]: --- Step: Switch to ios build lane ---
[19:24:14]: --------------------------------------
[19:24:14]: Cruising over to lane 'ios build' 🚖
[19:24:14]: build_model: true--->环境为测试环境
[19:24:14]: -----------------
[19:24:14]: --- Step: pwd ---
[19:24:14]: -----------------
[19:24:14]: $ pwd
[19:24:14]: ▸ /Users/XXX/jenkins/jenkins/workspace/CCC+fastlane/ios/fastlane
[19:24:14]: --------------------------------------------------------------------
[19:24:14]: Step: /usr/libexec/PlistBuddy -c 'Set :Debug true' ../CCC/Global/GlobalConfig.plist
[19:24:14]: --------------------------------------------------------------------
[19:24:14]: $ /usr/libexec/PlistBuddy -c 'Set :Debug true' ../CCC/Global/GlobalConfig.plist
[19:24:14]: ------------------------------------
[19:24:14]: --- Step: increment_build_number ---
[19:24:14]: ------------------------------------
Current version of project CCC is: 
    26

/Users/XXX/jenkins/jenkins/workspace/CCC+fastlane/ios
[19:24:14]: $ cd /Users/XXX/jenkins/jenkins/workspace/CCC\+fastlane/ios && agvtool next-version -all && cd -
[19:24:15]: ▸ Setting version of project CCC to:
[19:24:15]: ▸ 27.
[19:24:15]: ▸ Also setting CFBundleVersion key (assuming it exists)
[19:24:15]: ▸ Updating CFBundleVersion in Info.plist(s)...
[19:24:15]: ▸ Updated CFBundleVersion in "CCC.xcodeproj/../CCC/Main/Info.plist" to 27
[19:24:15]: ▸ /Users/XXX/jenkins/jenkins/workspace/CCC+fastlane/ios
[19:24:16]: ------------------------------
[19:24:16]: --- Step: get_build_number ---
[19:24:16]: ------------------------------
[19:24:16]: $ cd /Users/XXX/jenkins/jenkins/workspace/CCC\+fastlane/ios && agvtool what-version -terse
[19:24:16]: ▸ 27
[19:24:16]: --------------------------------
[19:24:16]: --- Step: get_version_number ---
[19:24:16]: --------------------------------
[19:24:16]: ------------------
[19:24:16]: --- Step: cert ---
[19:24:16]: ------------------

+---------------+--------------------------------------------------+
|                     Summary for cert 2.111.0                     |
+---------------+--------------------------------------------------+
| development   | false                                            |
| force         | false                                            |
| username      | CCC.cn                                   |
| team_id       | ....                                       |
| keychain_path | /Users/XXX/Library/Keychains/login.keychain-db |
| platform      | ios                                              |
+---------------+--------------------------------------------------+

[19:24:16]: Starting login with user 'CCC.cn'
Available session is not valid any more. Continuing with normal login.
[19:24:20]: Successfully logged in
[19:24:20]: Found the certificate JD.....7ad (iOS Distribution) which is installed on the local machine. Using this one.
[19:24:20]: Verifying the certificate is properly installed locally...
[19:24:20]: Successfully installed certificate JD.....7ad
[19:24:20]: Use signing certificate 'JD.....7ad' from now on!
[19:24:20]: true
[19:24:20]: adhock 进行打包
[19:24:20]: ------------------
[19:24:20]: --- Step: sigh ---
[19:24:20]: ------------------

+-------------------------------------+----------------+
|               Summary for sigh 2.111.0               |
+-------------------------------------+----------------+
| adhoc                               | true           |
| force                               | true           |
| developer_id                        | false          |
| development                         | false          |
| skip_install                        | false          |
| app_identifier                      | cn.CCC    |
| username                            | CCC.cn |
| team_id                             | ....     |
| ignore_profiles_with_different_name | false          |
| cert_id                             | JD.....7ad     |
| skip_fetch_profiles                 | false          |
| skip_certificate_verification       | false          |
| platform                            | ios            |
| readonly                            | false          |
+-------------------------------------+----------------+

[19:24:20]: Starting login with user 'CCC.cn'
Available session is not valid any more. Continuing with normal login.
[19:24:24]: Successfully logged in
[19:24:24]: Fetching profiles...
[19:24:25]: Verifying certificates...
[19:24:25]: Certificate for Provisioning Profile 'MAA' not available locally: 5AA7L94K9K, skipping this one...
[19:24:25]: No existing profiles found, that match the certificates you have installed locally! Creating a new provisioning profile for you
[19:24:27]: The name 'cn.CCC AdHoc' is already taken, using another one.
[19:24:27]: Creating new provisioning profile for 'cn.CCC' with name 'cn.CCC AdHoc 1545132267' for 'ios' platform
[19:24:28]: Downloading provisioning profile...
[19:24:28]: Successfully downloaded provisioning profile...
[19:24:28]: Installing provisioning profile...
/Users/XXX/jenkins/jenkins/workspace/CCC+fastlane/CZH_cers/AdHoc_cn.CCC.mobileprovision
[19:24:29]: Setting Provisioning Profile type to 'ad-hoc'
[19:24:29]: -----------------
[19:24:29]: --- Step: gym ---
[19:24:29]: -----------------
[19:24:29]: $ xcodebuild -showBuildSettings -workspace ./CCC.xcworkspace -scheme CCC
[19:24:32]: Command timed out after 3 seconds on try 1 of 4, trying again with a 6 second timeout...
[19:24:35]: Detected provisioning profile mapping: {:"cn.CCC"=>"cn.CCC AdHoc 1545132267"}

+-------------------------------------------------+----------------------------------------------------------+
|                                          Summary for gym 2.111.0                                           |
+-------------------------------------------------+----------------------------------------------------------+
| export_method                                   | ad-hoc                                                   |
| clean                                           | true                                                     |
| silent                                          | true                                                     |
| output_name                                     | CZH.v7.2.3.build27.2018.12.18                            |
| output_directory                                | ../CZH_ipas/                                             |
| export_options.provisioningProfiles.cn.CCC | cn.CCC AdHoc 1545132267                             |
| workspace                                       | ./CCC.xcworkspace                                   |
| scheme                                          | CCC                                                 |
| destination                                     | generic/platform=iOS                                     |
| build_path                                      | /Users/XXX/Library/Developer/Xcode/Archives/2018-12-18 |
| skip_package_ipa                                | false                                                    |
| result_bundle                                   | false                                                    |
| buildlog_path                                   | ~/Library/Logs/gym                                       |
| skip_profile_detection                          | false                                                    |
| xcode_path                                      | /Applications/Xcode.app                                  |
+-------------------------------------------------+----------------------------------------------------------+

[19:24:38]: ▸ Cleaning Pods/BBCyclingLabel [Debug]
[19:24:38]: ▸ ......
[19:24:38]: ▸ ......
[19:24:38]: ▸ ......
[19:29:30]: ▸ Running script '[CP] Embed Pods Frameworks'
[19:29:31]: ▸ Touching CCC.app
[19:29:33]: ▸ Signing /Users/XXX/Library/Developer/Xcode/DerivedData/CCC-hccilpreuzwhuwcgmaaciayxvfdf/Build/Intermediates.noindex/ArchiveIntermediates/CCC/InstallationBuildProductsLocation/Applications/CCC.app
[19:29:34]: ▸ Touching CCC.app.dSYM
[19:29:35]: ▸ Archive Succeeded
[19:29:36]: Generated plist file with the following values:
[19:29:36]: ▸ -----------------------------------------
[19:29:36]: ▸ {
[19:29:36]: ▸   "provisioningProfiles": {
[19:29:36]: ▸     "cn.CCC": "cn.CCC AdHoc 1545132267"
[19:29:36]: ▸   },
[19:29:36]: ▸   "method": "ad-hoc",
[19:29:36]: ▸   "signingStyle": "manual"
[19:29:36]: ▸ }
[19:29:36]: ▸ -----------------------------------------

[19:29:49]: Successfully exported and compressed dSYM file
[19:29:49]: Successfully exported and signed the ipa file:
[19:29:49]: /Users/XXX/jenkins/jenkins/workspace/CCC+fastlane/CZH_ipas/CZH.v7.2.3.build27.2018.12.18.ipa
[19:29:49]: Cruising back to lane 'ios start' 🚘

+------+---------------------------------------------------------------+-------------+
|                                  fastlane summary                                  |
+------+---------------------------------------------------------------+-------------+
| Step | Action                                                        | Time (in s) |
+------+---------------------------------------------------------------+-------------+
| 1    | default_platform                                              | 0           |
| 2    | git branch -l                                                 | 0           |
| 3    | git_branch                                                    | 0           |
| 4    | last_git_commit                                               | 0           |
| 5    | Switch to ios build lane                                      | 0           |
| 6    | pwd                                                           | 0           |
| 7    | /usr/libexec/PlistBuddy -c 'Set :Debug true' ../CCC/Glob | 0           |
| 8    | increment_build_number                                        | 1           |
| 9    | get_build_number                                              | 0           |
| 10   | get_version_number                                            | 0           |
| 11   | cert                                                          | 4           |
| 12   | sigh                                                          | 8           |
| 13   | gym                                                           | 320         |
| 14   | say                                                           | 2           |
+------+---------------------------------------------------------------+-------------+

[19:31:35]: fastlane.tools just saved you 7 minutes! 🎉
Finished: SUCCESS
</pre>
</details>

### Environment

<details>
  <pre><details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.14.1                                     |
| Ruby                        | 2.4.0                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.15.2 (Apple Git-101.1)        |
| Installation Source         | ~/.rvm/gems/ruby-2.4.0/bin/fastlane         |
| Host                        | Mac OS X 10.14.1 (18B75)                    |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.0/lib                |
| OpenSSL Version             | OpenSSL 1.1.0f  25 May 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.4.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself

#update_fastlane

default_platform(:ios)

platform :ios do


  before_all do |lane, options|
   
    # sh("pod install")
    sh("git branch -l")
    branchname = git_branch  
    lastcommit = last_git_commit
    UI.success "当前选择的分支为#{branchname}" 
    UI.success "最新的commit信息为#{lastcommit}"
  end

 desc "start"
  lane :for_pgyAdhoc do |options|
    #infomation 
    info = options[:des] 
    isDebug = options[:isDebug]
    ipapath =  build(isAdHoc: true, build_model: isDebug)
  end

  desc "archive lane"
  lane :build do |options|

    debug = false

    if options[:build_model] == true 

    puts "build_model: #{options[:build_model]}--->"
    debug = true
    elsif options[:build_model] == false
    puts "build_model: #{options[:build_model]}--->"
    debug = false
    else
    puts "build_model: #{options[:build_model]}--->c"
    debug = false
    end

    sh "pwd"
    sh"/usr/libexec/PlistBuddy -c 'Set :Debug #{debug}' ../ccc/Global/GlobalConfig.plist"
    increment_build_number  
    newbuild = get_build_number()
    newversion = get_version_number() 
    date = Time.new.strftime("%Y.%m.%d") 
    ipaname = "C.v#{newversion}.build#{newbuild}.#{date}.ipa"  
    path = "../C_ipas/"  
    ipapath = "#{path}#{ipaname}"
    cert(output_path: "../C_cers/")  

    puts "#{options[:isAdHoc]}"
     if options[:isAdHoc]
      puts "adhock build"
      sigh(adhoc: true, force: true, output_path: "../C_cers/") 
      gym(export_method: "ad-hoc",clean: true,silent: true,output_name: ipaname,output_directory: path) 
    else
      puts "生产 进行打包"
      sigh(output_path: "../C_cers/") 
      gym(export_method: "app-store",clean: true,silent: true,output_name: ipaname,output_directory: path)
    end
    
    ipapath

  end

 
error do |lane, exception, options|
  if options[:debug]
    puts "#{exception}"
  end
end

end


```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier("ccc") # The bundle identifier of your app
apple_id("ccc") # Your Apple email address

itc_team_id("ccc") # App Store Connect Team ID
team_id("ccc") # Developer Portal Team ID

# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.111.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                | Version | Update-Status      |
| --------------------- | ------- | ------------------ |
| fastlane-plugin-pgyer | 0.2.1   | 🚫 Update available |
| fastlane-plugin-pgyer | 0.2.1   | 🚫 Update available |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| executable-hooks          | 1.5.0        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.5        |
| bundler                   | 1.17.2       |
| io-console                | 0.4.6        |
| CFPropertyList            | 3.0.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.4.1        |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| dotenv                    | 2.5.0        |
| emoji_regex               | 0.1.1        |
| excon                     | 0.62.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.15.2       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.12.2       |
| fastimage                 | 2.1.4        |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| multi_json                | 1.13.1       |
| os                        | 1.0.0        |
| signet                    | 0.9.1        |
| googleauth                | 0.6.6        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2018.0812  |
| mime-types                | 3.2.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.0.5        |
| google-cloud-core         | 1.2.7        |
| google-cloud-storage      | 1.15.0       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.4.0        |
| rubyzip                   | 1.2.2        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.4.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.6.5        |
| tty-cursor                | 0.6.0        |
| tty-spinner               | 0.8.0        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.6.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
| fastlane-plugin-pgyer     | 0.2.1        |
</details>

*generated on:* **2018-12-17**
</details></pre>
</details>

**Comment by UZheng** [comment]
@yuewah @mouhao @squintanadev @shihkai-tony-wei @jianghui1  Here is the new issue, for recreating the new AdHoc profile in Apple Developer with different names .

**Comment by webdevbyjoss** [comment]
We have the same issue after Fastlane upgrade yesterday.

**Comment by Julien-Mialon** [comment]
Having the exact same issue with the command : 
`fastlane sigh -u user@company.com --app_identifier com.company.app --adhoc --team_name "Company" -o tmp`

Since we upgraded to 2.111.0, a provisioning profile is created each time a build is triggered on our CI

**Comment by joshdholtz** [comment]
Looking into this ^ 💪

**Comment by UZheng** [comment]
> Thanks for creating an issue that contains all the neccessary information @UZheng.
> 
> (I am still confused though as the output includes `adhock 进行打包` before `sigh` which is not present in your `Fastfile`. Where does this come from?)
> 
> Possibly a regression of #13907

This is a custom log in fastfile. For the convenience of reading, I deleted this command. Please ignore it.

**Comment by csknns** [comment]
The provision profiles of adhoc type are parsed as adhoc type, but before this(#13907) there was no way to distinguish them, so their type was set as appstore(workaround). In the method where the profiles where filtered by type the adhoc are filter as appstore, that results in an empty array, that causes the creation of new profiles.
I am creating a pull request for this.

**Comment by rubenstolk** [comment]
Same issue here since yesterday!

**Comment by joshdholtz** [comment]
Fixed with https://github.com/fastlane/fastlane/pull/13945

**Comment by benjarwar** [comment]
@joshdholtz we've started encountering this same issue, Fastlane is generating `adhoc` provisioning profiles with numbers at the end of their profile names. We've already upgraded to `2.112.0` and tried re-running `match`, but code signing still fails, because the profile that's downloaded for our app ID is like `match AdHoc com.foo.app-name 1545256864` but the cert is expecting `match AdHoc com.foo.app-name`.

Is there something else we need to do to trigger creation of the correctly named provisioning profile?
